reactor-core/src/test/java/reactor/core/publisher/FluxDematerializeTest.java

```diff
@@ -284,31 +240,68 @@ public void dematerialize() {
 		            .verifyComplete();
 	}
 
+	@Test
+	public void materializeDematerializeUnbounded() {
+		StepVerifier.create(Flux.just(1, 2, 3).materialize().dematerialize())
+		            .expectNext(1, 2, 3)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void materializeDematerializeRequestOneByOne() {
+		StepVerifier.create(Flux.just(1, 2, 3).materialize().dematerialize(), 0)
+		            .thenRequest(1)
+		            .expectNext(1)
+		            .thenRequest(1)
+		            .expectNext(2)
+		            .thenRequest(1)
+		            .expectNext(3)
+		            .expectNoEvent(Duration.ofMillis(50))
+		            .thenRequest(1)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void emissionTimingsAreGrouped() {
+		StepVerifier.withVirtualTime(() ->
+				Flux.interval(Duration.ofSeconds(1))
+				    .map(i -> "tick" + i)
+				    .take(5)
+				    .timestamp()
+				    .materialize()
+				    .<Tuple2<Long, String>>dematerialize()
+				    .timestamp()
+		)
+		            .thenAwait(Duration.ofSeconds(5))
+		            .thenConsumeWhile(tupleDematerialize -> {
+		            	long dematerializeTimestamp = tupleDematerialize.getT1();
+		            	long originalTimestamp = tupleDematerialize.getT2().getT1();
+
+			            System.out.println(tupleDematerialize.getT2().getT2() + ": " + originalTimestamp + ", " + dematerializeTimestamp);
```

Comment:
a leftover from debugging? Or we do need this logging?

Reply:
probably not that useful now that the test passes 😄